extensions/poetry/main.ts

```diff
@@ -146,74 +103,103 @@ Deno.exit(status.code);
 async function poetryCheckDryRun(
   subcommand: string,
   args: string[],
-): Promise<number> {
-  // Skip lockfile update on install, since it doesn't have the `--lock` flag.
-  if (subcommand !== "install") {
-    console.log(`[${green("phylum")}] Updating lockfile…`);
-
-    const status = PhylumApi.runSandboxed({
-      cmd: "poetry",
-      args: [subcommand, "-n", "--lock", ...args.map((s) => s.toString())],
-      exceptions: {
-        run: [
-          "./",
-          "/bin",
-          "/usr/bin",
-          "~/.pyenv",
-          "~/.local/bin/poetry",
-          "~/Library/Application Support/pypoetry",
-          "~/.local/share/pypoetry",
-        ],
-        write: [
-          "./",
-          "~/.cache/pypoetry",
-          "~/Library/Caches/pypoetry",
-          "~/.pyenv",
-        ],
-        read: [
-          "./",
-          "~/.cache/pypoetry",
-          "~/Library/Caches/pypoetry",
-          "~/.pyenv",
-          "~/Library/Preferences/pypoetry",
-          "~/.config/pypoetry",
-          "/etc/passwd",
-        ],
-        net: true,
-      },
-    });
-
-    // Ensure dry-run update was successful.
-    if (!status.success) {
-      console.error(`[${red("phylum")}] Lockfile update failed.\n`);
-      await abort(status.code ?? 255);
+) {
+  const result = PhylumApi.runSandboxed({
+    cmd: "poetry",
+    args: [
+      subcommand,
+      "--no-interaction",
+      "--dry-run",
+      ...args.map((s) => s.toString()),
+    ],
+    exceptions: {
+      run: [
+        "./",
+        "/bin",
+        "/usr/bin",
+        "~/.pyenv",
+        "~/.local/bin/poetry",
+        "~/Library/Application Support/pypoetry",
+        "~/.local/share/pypoetry",
+      ],
+      write: [
+        "~/.cache/pypoetry",
+        "~/Library/Caches/pypoetry",
+        "~/.pyenv",
+      ],
+      read: [
+        "./",
+        "~/.cache/pypoetry",
+        "~/Library/Caches/pypoetry",
+        "~/.pyenv",
+        "~/Library/Preferences/pypoetry",
+        "~/.config/pypoetry",
+        "/etc/passwd",
+      ],
+      net: true,
+    },
+    stdout: "piped",
+  });
+
+  // Ensure dry-run update was successful.
+  if (!result.success || result.stdout.length == 0) {
+    console.error(`[${red("phylum")}] Failed to determine new packages.\n`);
+    return result.code ?? 255;
+  }
+
+  // Parse dry-run output to look for new packages.
+  const packages = [];
+  const lines = result.stdout.split("\n");
+  for (const line of lines) {
+    const installing_text = "Installing ";
+    const installing_index = line.indexOf(installing_text);
+
+    // Filter lines unrelated to new packages.
+    if (installing_index === -1) {
+      continue;
```

Comment:
This method of finding and filtering lines based on the inclusion of `Installing ` is prone to false positives. That is, _any_ line with `Installing ` in it will pass through these checks and then may fail later when "Ensuring what we parsed is in a sensible format."

Indeed, testing this code resulted in one such error:

`[phylum] Invalid poetry output: Installing dependencies from lock file.`

This line is from the output of trying to run this command (shown executed without the extension):

```
../phylum-ci  6  17 on  main [?] is 📦 v0.24.1 via 🐍 v3.11.2 took 2s
❌ 125 ❯ poetry install --dry-run --sync --with test,qa
Installing dependencies from lock file

Package operations: 0 installs, 0 updates, 0 removals, 33 skipped

  • Installing attrs (22.2.0): Skipped for the following reason: Already installed
  • Installing backports-cached-property (1.0.2): Skipped for the following reason: Already installed
  • Installing cachetools (5.3.0): Skipped for the following reason: Already installed
  • Installing certifi (2022.12.7): Skipped for the following reason: Already installed
  • Installing cffi (1.15.1): Skipped for the following reason: Already installed
  • Installing chardet (5.1.0): Skipped for the following reason: Already installed
  • Installing charset-normalizer (3.0.1): Skipped for the following reason: Already installed
  • Installing colorama (0.4.6): Skipped for the following reason: Already installed
  • Installing commonmark (0.9.1): Skipped for the following reason: Already installed
  • Installing connect-markdown-renderer (2.0.1): Skipped for the following reason: Already installed
  • Installing cryptography (39.0.1): Skipped for the following reason: Already installed
  • Installing distlib (0.3.6): Skipped for the following reason: Already installed
  • Installing filelock (3.9.0): Skipped for the following reason: Already installed
  • Installing idna (3.4): Skipped for the following reason: Already installed
  • Installing iniconfig (2.0.0): Skipped for the following reason: Already installed
  • Installing packaging (23.0): Skipped for the following reason: Already installed
  • Installing mdurl (0.1.2): Skipped for the following reason: Already installed
  • Installing pluggy (1.0.0): Skipped for the following reason: Already installed
  • Installing pytest (7.2.1): Skipped for the following reason: Already installed
  • Installing pycparser (2.21): Skipped for the following reason: Already installed
  • Installing pyproject-api (1.5.0): Skipped for the following reason: Already installed
  • Installing tomli (2.0.1): Skipped for the following reason: Already installed
  • Installing markdown-it-py (2.2.0): Skipped for the following reason: Already installed
  • Installing rich (12.6.0): Skipped for the following reason: Already installed
  • Installing platformdirs (3.0.0): Skipped for the following reason: Already installed
  • Installing requests (2.28.2): Skipped for the following reason: Already installed
  • Installing tox (4.4.6): Skipped for the following reason: Already installed
  • Installing types-urllib3 (1.26.25.8): Skipped for the following reason: Already installed
  • Installing ruamel-yaml (0.17.21): Skipped for the following reason: Already installed
  • Installing urllib3 (1.26.14): Skipped for the following reason: Already installed
  • Installing virtualenv (20.19.0): Skipped for the following reason: Already installed
  • Installing types-requests (2.28.11.15): Skipped for the following reason: Already installed
  • Installing pygments (2.14.0): Skipped for the following reason: Already installed

Installing the current project: phylum (0.24.1)
```

Further, it is predicted that the same error would hit for the last line of the output:

`Installing the current project: phylum (0.24.1)`